### PR TITLE
Very Early mas_log

### PR DESCRIPTION
**Pull request by multimokia**
`renpy.log` (or `renpy.exports.log`) is extremely unreliable between init `-999` and `-995`. Sometimes it exists as one thing, sometimes it's not that, additionally because there is no logfile created internally at the time, we simply cannot trust this to log for us at these levels.

As such, I have moved the logging portions of `mas_utils` into `python early`, and it should be the first init block we have to run for MAS.

# Affected Members:
- `MASMacLog`
- `macLogOpen`
- `logcreate`
- `writelog`
- `wtf`
- `writestack`
- `logrotate`
- `tryparseint`
- `mas_mac_log_cache`
- `mas_log`
- `mas_log_open`

# Testing:
- Verify no crashes
- Verify logging still works
- Verify all functions exists as normal
- Verify logs still get rotated
- Verify logs work properly during early init levels (`-999` - `-995`)

### CODE REVIEW

**Review by ThePotatoGuy — CHANGES_REQUESTED** [review]
can the file be named `00utils`? Decent chance that we'll be adding more than just logging to here in the future.

**Review by ThePotatoGuy — DISMISSED** [review]

File: Monika After Story/game/00utils.rpy [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+                else:
+                    mode = "w"
+
+                if renpy.config.log_to_stdout:
+                    self.file = real_stdout
+
+                else:
+
+                    try:
+                        self.file = codecs.open(fn, mode, "utf-8")
```
Comment (Booplicate) [inline Monika After Story/game/00utils.rpy:60]: Should import `codecs`.

File: Monika After Story/game/00utils.rpy [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+                        pass
+
+                if self.append:
+                    self.write('')
+                    self.write('=' * 78)
+                    self.write('')
+
+                self.write("%s", time.ctime())
+                try:
+                    self.write("%s", platform.platform())
```
Comment (Booplicate) [inline Monika After Story/game/00utils.rpy:71]: Should import `platform`.

File: Monika After Story/game/00utils.rpy [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+        IN:
+            msg - message to log
+        """
+        writelog(msg)
+
+    def writestack():
+        """
+        Prints current stack to log
+        """
+        writelog("".join(traceback.format_stack()))
```
Comment (Booplicate) [inline Monika After Story/game/00utils.rpy:152]: Should import `traceback`.

File: Monika After Story/game/00utils.rpy [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+            logpath,
+            filename,
+            __numdelim,
+            __numformat.format(largest_num)
+        ]))
+        trydel(new_path)
+
+        # and copy our main file over
+        old_path = os.path.normcase(logpath + filename)
+        copyfile(old_path, new_path)
```
Comment (Booplicate) [inline Monika After Story/game/00utils.rpy:209]: `copyfile` is undefined.

File: Monika After Story/game/00utils.rpy [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+        largest_num += 1
+
+        # delete whatever file that is if it exists
+        new_path = os.path.normcase("".join([
+            logpath,
+            filename,
+            __numdelim,
+            __numformat.format(largest_num)
+        ]))
+        trydel(new_path)
```
Comment (Booplicate) [inline Monika After Story/game/00utils.rpy:205]: `trydel` is undefined.

**Review by Booplicate — CHANGES_REQUESTED** [review]

**Review by ThePotatoGuy — APPROVED** [review]

**Review by ThePotatoGuy — APPROVED** [review]

**Review by Booplicate — APPROVED** [review]
Can't test mac logs, but the code looks fine now.

**Review by Legendkiller21 — APPROVED** [review]

**Pull request merged**